astropy/time/formats.py

```diff
@@ -580,10 +585,11 @@ class TimeFromEpoch(TimeNumeric):
     def __init__(self, val1, val2, scale, precision,
                  in_subfmt, out_subfmt, from_jd=False):
         self.scale = scale
-        # Initialize the reference epoch (a single time defined in subclasses)
-        epoch = Time(self.epoch_val, self.epoch_val2, scale=self.epoch_scale,
-                     format=self.epoch_format)
-        self.epoch = epoch
+        if not hasattr(self.__class__, 'epoch'):
```

Comment:
How about using [`classproperty(lazy=True)`](https://docs.astropy.org/en/latest/api/astropy.utils.decorators.classproperty.html), as in
```
@classproperty(lazy=True)
def epoch(cls):
    return Time(cls.epoch_val, cls.epoch_val2, scale=cls.epoch_scale,
                format=cls.epoch_format)
```
(@embray has left us some very nice tricks!) It should be similar in speed to what you have here.

Reply:
p.s. Looking at the test failure, I'm fairly sure the test failures arise because by setting `self.__class__.epoch` in `TimeUnix` you also set the epoch for `TimeUnixTai`. You may not see that locally if somehow tests ran in different order (or you did not run all of them?)